space-concordia-robotics/robotics-prototype
# Feat command input 140

# Assignee Section

## Description
Made a command input field under the console, in order to be able to send custom commands to each module of the rover. Re-enabling an old removed feature. Slightly refactored some old code in order to make this work properly.

- [x] Added input text box and button into console.html file.
- [x] Added `cmd_input.js` file for housing the functions for the command input box and button.
- [x] Added 'cmd_input.css` file.
- [x] Modified some other `js` files in order to disable hotkeys on text box focus.

closes #140
closes #252

Approval from at least one software team lead is necessary before merging.

# Reviewer Section

## Local testing steps

1. Run: `startgui`, `rosgui`, `roslaunch local_rover.launch` in separate terminals
2. Plug in a teensy over usb to your computer and upload the `Arm.ino` sketch
3. Visit `localhost:5000`, activate the arm listener using the toggle button
4. Test sending regular manual mode arm movement commands (example: press `w`), observe the  builtin LED blink in response to receiving the command.
5. In the command input field type: `arm who`, press enter, overserve the LED blink in response.
6. Repeat step 5 but instead of pressing enter, click the send command button.

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
